### PR TITLE
Add documentation for knowhere project.

### PR DESCRIPTION
**Pull request by soothing-rain**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+If you wish to only use Knowhere within Milvus without changing any of the Knowhere source code, we suggest that you move to the [Milvus main project](https://github.com/milvus-io/milvus) and build Milvus directly, where Knowhere is then built implicitly during Milvus build.
+
+## System Requirements
+
+All Linux distributions are available for Knowhere development. However a majority of our contributor worked with Ubuntu or CentOS systems, with a small portion of Mac (both x86_64 and Apple Silicon) contributors. If you would like Knowhere to build and run on other distributions, you are more than welcome to file an issue and contribute!
+
+Here's a list of verified OS types where Knowhere can successfully build and run:
+
+- Debian/Ubuntu
```
Comment (cydrain) [inline README.md:17]: - Ubuntu 18.04

File: README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- CentOS
```
Comment (cydrain) [inline README.md:18]: remove CentOS

File: README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- MacOS (x86_64)
+- MacOS (Apple Silicon)
+
+## Building Knowhere From Source Code
+
+#### Install Dependencies
+
+```bash
+$ wget https://raw.githubusercontent.com/milvus-io/milvus/master/scripts/install_deps.sh
```
Comment (cydrain) [inline README.md:27]: we should use script under knowhere/build/install_deps.sh

**Comment by jaime0815** [comment]
https://github.com/milvus-io/knowhere/blob/main/build/install_deps.sh

File: README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+This document will help you to build the Knowhere repository from source code and to run unit tests. Please [file an issue](https://github.com/milvus-io/knowhere/issues/new) if there's a problem.
+
+## Introduction
+
+Knowhere is written in C++. It is an independent project that act as Milvus's internal core.
```
[the remark this answers is not in the document]
Comment (soothing-rain) [reply README.md:5]: Not sure what exactly should we put here, FAISS, HNSW, Annoy, NGT, SPTAG and NSG?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+- Ubuntu 18.04
+- MacOS (x86_64)
+- MacOS (Apple Silicon)
+
+## Building Knowhere From Source Code
+
+#### Install Dependencies
+
+```bash
+$ .build//install_deps.sh
```
Comment (cydrain) [inline README.md:26]: / duplicated

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (soothing-rain) [reply README.md:5]: As per offline discussion, will address in later version.

**Pull request merged**